api/src/main/java/org/openmrs/module/openconceptlab/UpdateServiceImpl.java

```diff
@@ -176,6 +176,15 @@ public Update getLastSuccessfulSubscriptionUpdate() {
 	}
 
 	@Override
+	public Boolean isLastUpdateSuccessful(){
+		if(getLastSuccessfulSubscriptionUpdate() == getLastUpdate()){

```

Comment:
It is hardly ever correct to rely on ==. You should be using equals. Also it doesn't make sense to use 'if' here since you return boolean anyway.